### PR TITLE
Refactor: Add namespace, enable sanitize builds

### PR DESCRIPTION
**Pull request by talregev**
- Add namespace
- Enable sanitize builds
- Refactor platform unittest external cmake

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -468,6 +468,8 @@ if(WIN32)
     add_custom_target(MsQuicEtw_HeaderBuild
         DEPENDS ${QUIC_BUILD_DIR}/inc/MsQuicEtw.h)
 
+    install(FILES ${QUIC_BUILD_DIR}/inc/MsQuicEtw.h DESTINATION include)
```
Comment (nibanks) [inline CMakeLists.txt:471]: Why is this necessary?

Comment (talregev) [reply CMakeLists.txt:471]: Because it looking MsQuicEtw.h in the installation header folder. 
Now msquic::msquic_platform take the detention and targets from cmake config, it also needed this header.

Comment (nibanks) [reply CMakeLists.txt:471]: What is 'it' in this context? I don't understand why anything outside of MsQuic internally would need this file.

Comment (talregev) [reply CMakeLists.txt:471]: quic_trance.h use this header. I will post the error in next comment.

**Comment by talregev** [comment]
Errors when remove line:
install(FILES ${QUIC_BUILD_DIR}/inc/MsQuicEtw.h DESTINATION include)

```
C:\Program Files\msquic\include\quic_trace.h(255,10): error C1083: Cannot open include file: 'MsQuicEtw.h': No such file or directory [D:\a\msquic\msquic\build_external\msquicplatformtest.vcxproj]
  (compiling source file '../src/platform/unittest/PlatformTest.cpp')
  
C:\Program Files\msquic\include\quic_trace.h(255,10): error C1083: Cannot open include file: 'MsQuicEtw.h': No such file or directory [D:\a\msquic\msquic\build_external\msquicplatformtest.vcxproj]
  (compiling source file '../src/platform/unittest/CryptTest.cpp')
  
C:\Program Files\msquic\include\quic_trace.h(255,10): error C1083: Cannot open include file: 'MsQuicEtw.h': No such file or directory [D:\a\msquic\msquic\build_external\msquicplatformtest.vcxproj]
  (compiling source file '../src/platform/unittest/main.cpp')
  
C:\Program Files\msquic\include\quic_trace.h(255,10): error C1083: Cannot open include file: 'MsQuicEtw.h': No such file or directory [D:\a\msquic\msquic\build_external\msquicplatformtest.vcxproj]
  (compiling source file '../src/platform/unittest/DataPathTest.cpp')
  
  TlsTest.cpp
C:\Program Files\msquic\include\quic_trace.h(255,10): error C1083: Cannot open include file: 'MsQuicEtw.h': No such file or directory [D:\a\msquic\msquic\build_external\msquicplatformtest.vcxproj]
  (compiling source file '../src/platform/unittest/TlsTest.cpp')
```

**Comment by talregev** [comment]
@nibanks From msquic api respective:
Did external user / library can define `QUIC_EVENTS_STDOUT` for msquic in their project? If yes, we should keep `MsQuicEtw.h` in the external include lib.
If not, I will remove the symbol `QUIC_EVENTS_STDOUT` and add `QUIC_EVENTS_STUB` manually, and remove  `MsQuicEtw.h` from the external include lib.

Let me know what do you think.

File: CMakeLists.txt (hunk 1 above)
Comment (nibanks) [reply CMakeLists.txt:471]: We shouldn't be using STDOUT by default. It's not performant.

Comment (talregev) [reply CMakeLists.txt:471]: ok, I will remove this define manually from external cmake, and remove this includes from external include folder (meaning remove this new line)

Comment (talregev) [reply CMakeLists.txt]: I remove this line.

**Comment by talregev** [comment]
@nibanks I remove the `MsQuicEtw.h` from external header folder. Please review again. Thank you!

**Comment by talregev** [comment]
@nibanks Can you approve the ci? Thank you for your time and effort for review my PR.

**Review by nibanks — APPROVED** [review]

**Comment by talregev** [comment]
@nibanks 
Thank you! Can you also release this with 2.4.*?

**Comment by nibanks** [comment]
> @nibanks Thank you! Can you also release this with 2.4.*?

Please create a cherry-pick PR for it, and run `.\scripts\update-version.ps1 -Part Patch`.

**Comment by talregev** [comment]
I run the script with errors. I am not sure why. I will create a PR without run this script.

**Comment by talregev** [comment]
```
PS C:\src\msquic> .\scripts\update-version.ps1 -Part Patch
Join-Path : A positional parameter cannot be found that accepts argument 'inc'. This failure might be caused by
applying the default parameter binding. You can disable the default parameter binding in $PSDefaultParameterValues by
setting $PSDefaultParameterValues["Disabled"] to be $true, and retry. The following default parameter was successfully
bound for this cmdlet when the error occurred: -ErrorAction
At C:\src\msquic\scripts\update-version.ps1:27 char:22
+ $MsQuicVerFilePath = Join-Path $RootDir "src" "inc" "msquic.ver"
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Join-Path], ParameterBindingException
    + FullyQualifiedErrorId : PositionalParameterNotFound,Microsoft.PowerShell.Commands.JoinPathCommand

Join-Path : A positional parameter cannot be found that accepts argument 'OneBranch.Package.yml'. This failure might
be caused by applying the default parameter binding. You can disable the default parameter binding in
$PSDefaultParameterValues by setting $PSDefaultParameterValues["Disabled"] to be $true, and retry. The following
default parameter was successfully bound for this cmdlet when the error occurred: -ErrorAction
At C:\src\msquic\scripts\update-version.ps1:28 char:24
+ ... teVPackFilePath = Join-Path $RootDir ".azure" "OneBranch.Package.yml"
+                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Join-Path], ParameterBindingException
    + FullyQualifiedErrorId : PositionalParameterNotFound,Microsoft.PowerShell.Commands.JoinPathCommand

Join-Path : A positional parameter cannot be found that accepts argument 'package-nuget.ps1'. This failure might be
caused by applying the default parameter binding. You can disable the default parameter binding in
$PSDefaultParameterValues by setting $PSDefaultParameterValues["Disabled"] to be $true, and retry. The following
default parameter was successfully bound for this cmdlet when the error occurred: -ErrorAction
At C:\src\msquic\scripts\update-version.ps1:29 char:21
+ $NugetPackageFile = Join-Path $RootDir "scripts" "package-nuget.ps1"
+                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Join-Path], ParameterBindingException
    + FullyQualifiedErrorId : PositionalParameterNotFound,Microsoft.PowerShell.Commands.JoinPathCommand

Join-Path : A positional parameter cannot be found that accepts argument 'package-distribution.ps1'. This failure
might be caused by applying the default parameter binding. You can disable the default parameter binding in
$PSDefaultParameterValues by setting $PSDefaultParameterValues["Disabled"] to be $true, and retry. The following
default parameter was successfully bound for this cmdlet when the error occurred: -ErrorAction
At C:\src\msquic\scripts\update-version.ps1:30 char:21
+ ... ibutionFile = Join-Path $RootDir "scripts" "package-distribution.ps1"
+                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Join-Path], ParameterBindingException
    + FullyQualifiedErrorId : PositionalParameterNotFound,Microsoft.PowerShell.Commands.JoinPathCommand

Join-Path : A positional parameter cannot be found that accepts argument 'distribution'. This failure might be caused
by applying the default parameter binding. You can disable the default parameter binding in $PSDefaultParameterValues
by setting $PSDefaultParameterValues["Disabled"] to be $true, and retry. The following default parameter was
successfully bound for this cmdlet when the error occurred: -ErrorAction
At C:\src\msquic\scripts\update-version.ps1:31 char:22
+ ... meworkInfoFile = Join-Path $RootDir "src" "distribution" "Info.plist"
+                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Join-Path], ParameterBindingException
    + FullyQualifiedErrorId : PositionalParameterNotFound,Microsoft.PowerShell.Commands.JoinPathCommand

Join-Path : A positional parameter cannot be found that accepts argument 'write-versions.ps1'. This failure might be
caused by applying the default parameter binding. You can disable the default parameter binding in
$PSDefaultParameterValues by setting $PSDefaultParameterValues["Disabled"] to be $true, and retry. The following
default parameter was successfully bound for this cmdlet when the error occurred: -ErrorAction
At C:\src\msquic\scripts\update-version.ps1:33 char:22
+ ... VersionsWriteFile = Join-Path $RootDir "scripts" "write-versions.ps1"
+                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Join-Path], ParameterBindingException
    + FullyQualifiedErrorId : PositionalParameterNotFound,Microsoft.PowerShell.Commands.JoinPathCommand

Select-String : Cannot bind argument to parameter 'Path' because it is null.
At C:\src\msquic\scripts\update-version.ps1:38 char:34
+ $VerMajor = (Select-String -Path $MsQuicVerFilePath "#define VER_MAJO ...
+                                  ~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Select-String], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.SelectStrin
   gCommand

Select-String : Cannot bind argument to parameter 'Path' because it is null.
At C:\src\msquic\scripts\update-version.ps1:39 char:34
+ $VerMinor = (Select-String -Path $MsQuicVerFilePath "#define VER_MINO ...
+                                  ~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Select-String], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.SelectStrin
   gCommand

Select-String : Cannot bind argument to parameter 'Path' because it is null.
At C:\src\msquic\scripts\update-version.ps1:40 char:34
+ $VerPatch = (Select-String -Path $MsQuicVerFilePath "#define VER_PATC ...
+                                  ~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Select-String], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.SelectStrin
   gCommand

Current version: ..
    New version: 0.0.1
Get-Content : Cannot bind argument to parameter 'Path' because it is null.
At C:\src\msquic\scripts\update-version.ps1:56 char:14
+ (Get-Content $MsQuicVerFilePath) `
+              ~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Get-Content], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.GetContentC
   ommand

Get-Content : Cannot bind argument to parameter 'Path' because it is null.
At C:\src\msquic\scripts\update-version.ps1:61 char:14
+ (Get-Content $CreateVPackFilePath) `
+              ~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Get-Content], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.GetContentC
   ommand

Get-Content : Cannot bind argument to parameter 'Path' because it is null.
At C:\src\msquic\scripts\update-version.ps1:70 char:14
+ (Get-Content $NugetPackageFile) `
+              ~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Get-Content], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.GetContentC
   ommand

Get-Content : Cannot bind argument to parameter 'Path' because it is null.
At C:\src\msquic\scripts\update-version.ps1:73 char:14
+ (Get-Content $FrameworkInfoFile) `
+              ~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Get-Content], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.GetContentC
   ommand

Get-Content : Cannot bind argument to parameter 'Path' because it is null.
At C:\src\msquic\scripts\update-version.ps1:76 char:14
+ (Get-Content $DistributionFile) `
+              ~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Get-Content], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.GetContentC
   ommand

Get-Content : Cannot bind argument to parameter 'Path' because it is null.
At C:\src\msquic\scripts\update-version.ps1:79 char:14
+ (Get-Content $VersionsWriteFile) `
+              ~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Get-Content], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.GetContentC
   ommand
```

**Comment by nibanks** [comment]
Make sure you're using a newer version of powershell (7+).

**Comment by talregev** [comment]
> Make sure you're using a newer version of powershell (7+).

@nibanks 
I upgrade to poweshell 7, and run the script, and create a PR:
https://github.com/microsoft/msquic/pull/4701
